akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowPrefixAndTailSpec.scala

```diff
@@ -192,6 +192,23 @@ class FlowPrefixAndTailSpec extends AkkaSpec {
       upsub.expectCancellation()
     }
 
+    "work even if tail subscriber arrives after substream completion" in {
+      val pub = TestPublisher.manualProbe[Int]()
+      val sub = TestSubscriber.manualProbe[Int]()
+
+      val f = Source(pub).prefixAndTail(1).runWith(Sink.head)
+      val s = pub.expectSubscription()
+      s.sendNext(0)
+
+      val (_, tail) = Await.result(f, 3.seconds)
+
+      val tailPub = tail.runWith(Sink.publisher)
+      s.sendComplete()
+
+      tailPub.subscribe(sub)
+      sub.expectSubscriptionAndComplete()
```

Comment:
Ok, I think I see how it wouldn't hit this path... There is a test for completely empty stream, but that's slightly different.